lib/Sema/CSRanking.cpp

```diff
@@ -33,23 +33,11 @@ using namespace constraints;
 STATISTIC(NumDiscardedSolutions, "Number of solutions discarded");
 
 void ConstraintSystem::increaseScore(ScoreKind kind, unsigned value) {
-  if (isForCodeCompletion()) {
-    switch (kind) {
-    case SK_NonDefaultLiteral:
-      // Don't increase score for non-default literals in expressions involving
-      // a code completion. In the below example, members of EnumA and EnumB
-      // should be ranked equally:
-      //   func overloaded(_ x: Float, _ y: EnumA) {}
-      //   func overloaded(_ x: Int, _ y: EnumB) {}
-      //   func overloaded(_ x: Float) -> EnumA {}
-      //   func overloaded(_ x: Int) -> EnumB {}
-      //
-      //   overloaded(1, .<complete>) {}
-      //   overloaded(1).<complete>
-      return;
-    default:
-      break;
-    }
+  if (isForCodeCompletion() && kind >= SK_SyncInAsync) {
```

Comment:
Should this be done locally only for locations that involve code completion token in some way (calls with code completion in argument or member references with code completion token?

It's bad that we didn't use `SK_NonDefaultLiteral` unconditionally today because expressions like `1 + 2 + #^COMPLETE^#` - would produce all possible permutations of non-default types like `Int32, UInt16, Double` etc. even for the first `+` which degrades performances significantly for filtering.